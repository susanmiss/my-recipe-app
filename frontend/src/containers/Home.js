import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import Logo from '../components/logo.jpg'


const App = () => {
  const [recipes, setRecipes, id] = useState([]);


  const fetchRecipes = () => {
    axios.get(`${process.env.REACT_APP_API}/list`)
      .then(response => {
        console.log(response)
        setRecipes(response.data)
      })
      .catch(error => alert('Error fetching recipes'))
  }

  useEffect(() => {
    fetchRecipes()
  }, [])

  const deleteConfirm = (id) => {
    let answer = window.confirm('Are you sure you want to delete this delicius recipe?')
    if (answer) {
      deleteRecipe(id)
    }
  }

  const deleteRecipe = (id) => {
    axios
      .delete(`${process.env.REACT_APP_API}/${id}`)
      .then(response => {
        alert(response.data.message)
        fetchRecipes()
      })
      .catch(error => alert('error deleting this amazing recipe'))
  }


  return (
    <div className="container pb-5">

      <br />
      <br />
      <h1 style={{ textAlign: 'center' }}>Getting fat in lockdown</h1>
      <h2>Share only FAT recipes that can make us very fat :)</h2>
      <br />
      <br />
      <img src={Logo} alt="logo" className="logo-img" />
      <hr />
      {/* {JSON.stringify(posts)} */}
      {
        recipes.map((recipe, i) => (
          <div className="row" key={recipe.id} style={{ borderBottom: '1px solid silver' }}>
            <div className="col pt-3 pb-2">

              <div className='row'>

                <div className="col-md-10">
                <Link to={`/recipe/${recipe.id}`}> <h2>{recipe.recipeTitle}</h2></Link>
           
                </div>

               
                  <div className="col-md-2">
                  <Link to={`recipe/update/${recipe.id}`} className="btn btn-sm btn-outline-warning">
                      Update Recipe
                  </Link>
                  <br />
                  <br />

                  <button onClick={() => deleteConfirm(recipe.id)} className="btn btn-sm btn-outline-danger ml-1">Delete Recipe</button>

                  </div>
              

              </div>
            </div>
          </div>
        ))
      }
    </div>
  )
}


export default App;

