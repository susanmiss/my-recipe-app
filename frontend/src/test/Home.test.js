import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import Home from '../containers/Home'


describe("<Home />", () => {
    
  it('mounts to DOM', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Home />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
  it('renders correctly', () => {
      const tree = renderer
          .create(<Home />)
          .toJSON();
      expect(tree).toMatchSnapshot();
  });
  let wrapper;
  beforeEach(() => wrapper = shallow(<Home />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div />', () => { 
      expect(wrapper.find('div').length).toEqual(1);
  });

 
})