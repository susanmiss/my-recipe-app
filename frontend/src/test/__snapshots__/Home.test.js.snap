// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<Home /> renders correctly 1`] = `
<div
  className="container pb-5"
>
  <br />
  <br />
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Getting fat in lockdown
  </h1>
  <h2>
    Share only FAT recipes that can make us very fat :)
  </h2>
  <br />
  <br />
  <img
    alt="logo"
    className="logo-img"
    src="logo.jpg"
  />
  <hr />
</div>
`;
